### DISCUSSION TITLE
Restore Borderless Mode for Linux (& possibly MacOS)

### DISCUSSION THREAD

**Discussion by limefps** [ppy/osu#28221] [Ideas]
The removal of the Borderless option on Linux leads to forced cursor confinement:

As there no longer is a Borderless option in 2024.519 (on Linux and MacOS, although I don't know if this would also apply to the latter), and the cursor is confined in Fullscreen mode, doing something on a second monitor now requires a full alt-tab, whereas previously with Borderless and confine cursor set to Gameplay or Never, doing things on a second monitor was a significantly better experience.

Quick note on workarounds: most compositors & DWMs support manually initiating Borderless Windowed mode for Windowed apps. This is possible in KWin (for example), but it forcibly minimizes the game when clicking on a different window (this is SDL behaviour, and not wanted)
If you are running into this same issue when trying this workaround, you can try setting 'export SDL_VIDEO_MINIMIZE_ON_FOCUS_LOSS=0' in /etc/environment and rebooting, but this really is a suboptimal user experience and at least with KWin borderless needs to be manually enabled each time the game is launched

**Comment by umuod** [comment]
Hello. Regarding the so-called "removal" of borderless option on Linux and MacOS that came along with the upgrade to SDL3 in the recent update, see: https://github.com/ppy/osu-framework/pull/6234

As for me who uses KDE Wayland and launches osu!lazer through a .desktop file. Firstly, regarding the game minimizing on focus loss. It seems to be by default set to true inside the framework code and had been so for several patches now. According to [the SDL doc regarding environment variables](https://www.libsdl.org/release/SDL-1.2.15/docs/html/sdlenvvars.html), I simply added `env SDL_VIDEO_MINIMIZE_ON_FOCUS_LOSS=0` before the the execute file path in the .desktop file. Which I think is a better option if you want to keep applications minimizing on focus loss per-application case. You can simpy save the .desktop file after and launch it.

And secondly, regarding the cursor being confined inside the game window. I got it too after updating to the recent version. If you are on X11 you could consider adding `env SDL_MOUSE_RELATIVE=0` from the aforementioned SDL doc regarding env vars in a similar way to `SDL_VIDEO_MINIMIZE_ON_FOCUS_LOSS`. If that doesn't work, my workaround for KDE Wayland was setting the game to Windowed mode.  And setting the C`onfine mouse cursor to window` option in the osu!lazer settings under Mouse to `never` (this setting is not applicable during full-screen mode) and ask KWin to fullscreen osu!lazer (can be done automatically through KWin Window Behaviour). I have been playing for the past few hours and noticed no change in latency or performance on my part.

I'm unsure about MacOS or Linux Gnome users but it doesn't look like I had any issue with this at all. And once again, I think it was an upstream change so I'm not sure there is much the devs could do about it. Lmk if you need additional help to get it work.
